config/daemonconfig/fuse.go

```diff
@@ -94,9 +95,29 @@ func (c *FuseDaemonConfig) DumpString() (string, error) {
 	return DumpConfigString(c)
 }
 
+// We can set the Backend tag to "-" so that Backend Config does not write to local files.
+// But it needs to be compatible with the old version,
+// and it is not possible to control whether Backend Config is written to the local file through tag.
+// Therefore, a deep copy of the Config information is made here,
+// and the Backend Config is set to null.
 func (c *FuseDaemonConfig) DumpFile(f string) error {
 	if err := os.MkdirAll(path.Dir(f), 0755); err != nil {
 		return err
 	}
+	if config.IsBackendSourceEnabled() {
+		configJSON, err := json.Marshal(c)
+		if err != nil {
+			return err
+		}
+
+		var configMap map[string]interface{}
+		err = json.Unmarshal(configJSON, &configMap)
```

Comment:
The process of Marshal and Unmarshal is a bit strange, in fact we only need to set some sensitive fields to an empty string (such as `BackendConfig.Auth`, `BackendConfig.RegistryToken`, `BackendConfig.AccessKeyID`, etc.). Or we can do nothing and let users not fill in sensitive fields in config.json.

In addition, we should not completely clear the `BackendConfig`, otherwise we will not be able to restore nydusd after the snapshotter restarts, please make sure that your PR still works in this case.

Reply:
@imeoer  I have now modified my plan and only these sensitive fields (such as BackendConfig.Auth, BackendConfig.RegistryToken, BackendConfig.AccessKeyID, etc.). will not be written to the local file.